applicationset/generators/matrix.go

```diff
@@ -50,10 +50,17 @@ func (m *MatrixGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.App
 	if err != nil {
 		return nil, err
 	}
+	requiresInterpolation := false // Try to generate 2nd generator's params without interpolation
+	g1, err := m.getParams(appSetGenerator.Matrix.Generators[1], appSet, nil)
+	if err != nil || g1 == nil {
```

Comment:
A little unusual, but I like the simplicity. :-) Wanna add tests to show that the generators aren't run extra times?

Reply:
Hey @crenshaw-dev thanks for the review!! Yes for sure, I'm currently in the middle of writing some tests that represent just that.

Reply:
@crenshaw-dev - The test should be good to go. The test was initially causing many issues for some reason, but came to the realisation that for the "Interpolated Generator" (ie. like below), the 1st `GenerateParams` would end up returning nil/no-op (which makes sense according to the code). So managed to get that factored into the test-case.

```yaml
	interpolatedGitGenerator := &argoprojiov1alpha1.GitGenerator{
		RepoURL:  "RepoURL",
		Revision: "Revision",
		Files: []argoprojiov1alpha1.GitFileGeneratorItem{
			{Path: "examples/git-generator-files-discovery/cluster-config/{{name}}.json"},
		},
	}
```

Reply:
## Test Summary

### Test-Setup
Matrix generator w/:
1) Cluster-Generator - dev + prod Clusters
2) Git Generator - With single git-files path entry 

#### Test 1 - Interpolated Generator
- 3 "GenerateParams" calls total - 1 No-Op, 1-Dev, 1-Prod

#### Test 2 - NonInterpolated Generator
- 1 "GenerateParams" call - 1 regular call


Anyways, let me know if you have any feedback/recommendations. Thanks!

Reply:
@crenshaw-dev - Let me know if there's anything else you need on the PR. Thanks!